build_functions/CMakeLists.txt

```diff
@@ -193,3 +193,43 @@ function(compileTargetAsC11 theTarget)
     set_target_properties(${theTarget} PROPERTIES CXX_STANDARD 11)
   endif()
 endfunction()
+
+# variable to store list of libs that must be checked for reals
+set(LIBS_TO_BE_REAL_CHECKED CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+
+# overloading the add_library function to store name of library if it meets criteria
+function (add_library())
+    _add_library(${ARGV})
+    get_target_property(lib_type ${ARGV0} TYPE)
+    if( (NOT ${CMAKE_CURRENT_LIST_DIR} MATCHES "/deps/") AND # ignore dependency libs
+        (NOT ${ARGV0} MATCHES "_ut_|_int_|_perf_") AND # ignore test libs
```

Comment:
I'm not sure I understand your concern. Adding "_int_" and "_perf_" make sure that those libs are not checked for reals. Since real checking happens in jobs where int tests are built, we need "_int_" here to exclude those libs from getting checked.